contracts/src/RealitioForeignArbitrationProxyWithAppeals.sol

```diff
@@ -0,0 +1,604 @@
+    function acknowledgeArbitration(bytes32 _questionID) external override onlyAmb onlyHomeChain onlyHomeProxy {
+        uint256 arbitrationID = uint256(_questionID);
+        Arbitration storage arbitration = arbitrations[arbitrationID];
+        require(arbitration.status == Status.Requested, "Invalid arbitration status");
+
+        uint256 arbitrationCost = arbitrator.arbitrationCost(arbitratorExtraData);
+
+        if (arbitration.deposit < arbitrationCost) {
+            arbitration.status = Status.Failed;
+
+            emit ArbitrationFailed(arbitrationID);
+        } else {
+            // At this point, arbitration.deposit is guaranteed to be greater than or equal to the arbitration cost.
+            uint256 remainder = arbitration.deposit - arbitrationCost;
+
+            uint256 disputeID =
+                arbitrator.createDispute{value: arbitrationCost}(NUMBER_OF_CHOICES_FOR_ARBITRATOR, arbitratorExtraData);
+            externalIDtoLocalID[disputeID] = arbitrationID;
+            arbitration.status = Status.Created;
+            arbitration.deposit = 0;
+            arbitration.disputeID = disputeID;
+            arbitration.rounds.push();
+
+            if (remainder > 0) {
+                arbitration.requester.send(remainder);
+            }
+
+            emit ArbitrationCreated(arbitrationID, disputeID);
+        }
+    }
+
+    /**
+     * @notice Cancels the arbitration.
+     * @param _questionID The ID of the question.
+     */
+    function cancelArbitration(bytes32 _questionID) external override onlyAmb onlyHomeChain onlyHomeProxy {
+        uint256 arbitrationID = uint256(_questionID);
+        Arbitration storage arbitration = arbitrations[uint256(_questionID)];
+        require(arbitration.status == Status.Requested, "Invalid arbitration status");
+
+        arbitration.requester.send(arbitration.deposit);
+
+        delete arbitrations[arbitrationID];
+
+        emit ArbitrationCanceled(arbitrationID);
+    }
+
+    /**
+     * @notice Cancels the arbitration in case the dispute could not be created.
+     * @param _questionID The ID of the question.
+     */
+    function handleFailedDisputeCreation(bytes32 _questionID) external onlyIfInitialized {
+        uint256 arbitrationID = uint256(_questionID);
+        Arbitration storage arbitration = arbitrations[arbitrationID];
+        require(arbitration.status == Status.Failed, "Invalid arbitration status");
+
+        bytes4 methodSelector = IHomeArbitrationProxy(0).receiveArbitrationFailure.selector;
+        bytes memory data = abi.encodeWithSelector(methodSelector, _questionID);
+        amb.requireToPassMessage(homeProxy, data, amb.maxGasPerTx());
+
+        arbitration.requester.send(arbitration.deposit);
+
+        delete arbitrations[arbitrationID];
+
+        emit ArbitrationCanceled(arbitrationID);
+    }
+
+    /** @notice Takes up to the total amount required to fund an answer. Reimburses the rest. Creates an appeal if at least two answers are funded.
```

Comment:
Is there any possibility of ever having more than 2 answers fully funded?

Reply:
We the 2 are funded the round is switched, so no. Many answers can be funded partially though